yasnippet.el

```diff
@@ -3467,17 +3467,23 @@ Move the overlays, or create them if they do not exit."
              ;; (overlay-put ov 'evaporate t)
              (overlay-put ov 'modification-hooks '(yas--on-protection-overlay-modification)))))))
 
-(defun yas--on-protection-overlay-modification (_overlay after? _beg _end &optional _length)
+(defvar yas--buffer-pre-modification)
```

Comment:
If we save only the length, then we'd miss modifications that replaced actual text with something of equal length, no?